clippy_lints/src/operators/arithmetic_side_effects.rs

```diff
@@ -184,6 +184,35 @@ impl ArithmeticSideEffects {
         }
     }
 
+    /// There are some integer methods like `wrapping_div` that will panic depending on the
+    /// provided input.
+    fn manage_method_call<'tcx>(
+        &mut self,
+        args: &[hir::Expr<'tcx>],
+        cx: &LateContext<'tcx>,
+        instance: &hir::Expr<'tcx>,
+        ps: &hir::PathSegment<'tcx>,
+    ) {
+        const METHODS: &[&str] = &["saturating_div", "wrapping_div", "wrapping_rem", "wrapping_rem_euclid"];
+        let Some(arg) = args.first() else { return; };
```

Comment:
```suggestion
```

Reply:
Right now only methods with one input are considered and in case of a method without inputs, the move of this branch will incur the evaluation of potentially unnecessary `constant_simple`, `is_integral`, and `METHODS.iter().copied().all(|method| method != ps.ident.as_str())` calls.